### PR TITLE
Enhancement/improve bcrypt

### PR DESCRIPTION
**Pull request by Vec7or**
#488 I added a environment variable to specify the bcrypt-work-factor. There are still two To Do's within the code-base because I do not know how you want me to handle the logging.

### CODE REVIEW

File: docs-core/src/main/java/com/sismics/docs/core/dao/UserDao.java [hunk 1]
```diff
@@ -278,7 +279,22 @@ public void delete(String username, String userId) {
...
-        return BCrypt.withDefaults().hashToString(10, password.toCharArray());
+        int bcryptWork = Constants.DEFAULT_BCRYPT_WORK;
+        String envBcryptWork = System.getenv(Constants.BCRYPT_WORK_ENV);
+        if (envBcryptWork != null) {
+            try {
+                int envBcryptWorkInt = Integer.parseInt(envBcryptWork);
+                if (envBcryptWorkInt >= 4 && envBcryptWorkInt <= 31) {
+                   bcryptWork = envBcryptWorkInt;
+                } else {
+                    //ToDo: Log warning? (out of scope)
```
Comment (jendib) [inline docs-core/src/main/java/com/sismics/docs/core/dao/UserDao.java:290]: Log using slf4j logger (please see FileService.java for example) as "warning" level and fallback to default bcrypt work.

File: docs-core/src/main/java/com/sismics/docs/core/dao/UserDao.java [hunk 2]
```diff
@@ -278,7 +279,22 @@ public void delete(String username, String userId) {
...
+                }
+            } catch (NumberFormatException e) {
+                //ToDo: Log warning? (no number)
```
Comment (jendib) [inline docs-core/src/main/java/com/sismics/docs/core/dao/UserDao.java:293]: Same as above

File: docs-core/src/main/java/com/sismics/docs/core/dao/UserDao.java [hunk 3]
```diff
@@ -278,7 +279,22 @@ public void delete(String username, String userId) {
...
+            }
+        }
+        System.out.println(bcryptWork);
```
Comment (jendib) [inline docs-core/src/main/java/com/sismics/docs/core/dao/UserDao.java:296]: Please remove

**Review by jendib — CHANGES_REQUESTED** [review]
I added a few comments if you can resolve this. Thanks.

File: docs-core/src/main/java/com/sismics/docs/core/dao/UserDao.java (hunk 3 above)
Comment (Vec7or) [reply docs-core/src/main/java/com/sismics/docs/core/dao/UserDao.java]: I must have overlooked that 😄.

**Comment by Vec7or** [comment]
Adjusted the code according to the review.

**Review by jendib — APPROVED** [review]

**Pull request merged**